Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
> I think HEAD should be updated to stable when we can detect it's too outdated i.e. when stable have changed since we last installed HEAD

It's tough; I'm not sure how many people will want to always follow HEAD vs. just upgrade to stable. Would appreciate some other @Homebrew/maintainers thoughts here.

> What do you mean?

Don't care about `LinkedKegs` and just the `opt/foo` link i.e. don't care about `brew link` status.



> well, maybe this speaks for itself that indeed we need to put the method into Dependency. 😃

😁 I agree!